acceptance/framework/consul/helm_cluster.go

```diff
@@ -297,12 +298,51 @@ func (h *HelmCluster) CreatePortForwardTunnelToResourcePort(t *testing.T, resour
 		require.NoError(r, tunnel.ForwardPortE(t))
 	})
 
+	doneChan := make(chan bool)
+
 	t.Cleanup(func() {
-		tunnel.Close()
+		close(doneChan)
 	})
 
+	go h.monitorPortForwardedServer(t, localPort, tunnel, doneChan, resourceName, remotePort)
+
 	return fmt.Sprintf("127.0.0.1:%d", localPort)
+}
 
+func (h *HelmCluster) monitorPortForwardedServer(t *testing.T, port int, tunnel *terratestk8s.Tunnel, doneChan chan bool, resourceName string, remotePort int) {
+	ticker := time.NewTicker(1 * time.Second)
+
+	for {
+		select {
+		case <-doneChan:
+			logger.Log(t, "stopping monitor of the port-forwarded server")
+			tunnel.Close()
+			return
+		case <-ticker.C:
+			conn, err := net.Dial("tcp", fmt.Sprintf("127.0.0.1:%d", port))
+			logger.Log(t, "lost connection to port-forwarded server", "port", port)
```

Comment:
Is this log message true 100% of the time?  (Or am I miss-understanding this) The ticker is running every 1 second, you establish a connection, check if there was an error, if there was an error re-establish tunnel.

Reply:
Great question. No it's not right. It should only be printed if there's an error. Great catch and thank you!!